splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -2148,276 +2149,218 @@ public static void SHOW_CREATE_TABLE(String schemaName, String tableName, Result
+
+                String tmpStr;
+                isExternal = "EXTERNAL ";
+                tmpStr = td.getCompression();
+                if (tmpStr != null && !tmpStr.equals("none"))
+                    extTblString.append("\nCOMPRESSED WITH " + tmpStr);
+
+                // Partitioned Columns
+                boolean firstCol = true;
+                for (ColumnDescriptor col: partitionColumns) {
+                    extTblString.append(firstCol ? "\nPARTITIONED BY (\"" + col.getColumnName()+"\"" : ",\"" + col.getColumnName()+"\"");
+                    firstCol = false;
+                }
+
+                if (!firstCol)
+                    extTblString.append(")");
+
+                // Row Format
+                if (td.getDelimited() != null || td.getLines() != null) {
+                    extTblString.append("\nROW FORMAT DELIMITED");
+                    if ((tmpStr = td.getDelimited()) != null)
+                        extTblString.append(" FIELDS TERMINATED BY ''" + tmpStr + "''");
+                    if ((tmpStr = td.getEscaped()) != null)
+                        extTblString.append(" ESCAPED BY ''" + tmpStr + "''");
+                    if ((tmpStr = td.getLines()) != null)
+                        extTblString.append(" LINES TERMINATED BY ''" + tmpStr + "''");
+                }
+                // Storage type
+                if ((tmpStr = td.getStoredAs()) != null) {
+                    extTblString.append("\nSTORED AS ");
+                    switch (tmpStr) {
+                        case "T":
+                            extTblString.append("TEXTFILE");
+                            break;
+                        case "P":
+                            extTblString.append("PARQUET");
+                            break;
+                        case "A":
+                            extTblString.append("AVRO");
+                            break;
+                        case "O":
+                            extTblString.append("ORC");
+                            break;
+                        default:
+                            throw new SQLException("Invalid stored format");
                     }
                 }
+                // Location
+                if ((tmpStr = td.getLocation()) != null) {
+                    extTblString.append("\nLOCATION ''" + tmpStr + "''");
+                }
+            }//End External Table
+            else if (td.getTableType() == TableDescriptor.VIEW_TYPE) {
+                //Target table is a View
+                throw ErrorState.LANG_INVALID_OPERATION_ON_VIEW.newException("SHOW CREATE TABLE", "\"" + schemaName + "\".\"" + tableName + "\"");
+            } else if (td.getTableType() == TableDescriptor.SYSTEM_TABLE_TYPE) {
+                //Target table is a system table
+                throw ErrorState.LANG_NO_USER_DDL_IN_SYSTEM_SCHEMA.newException("SHOW CREATE TABLE", schemaName);
+            }
+            // Get column list, and write DDL for each column.
```

Comment:
The new code here might introduce a behavior change. It's tricky. In the old query, predicate on table type is `T.TABLETYPE IN ('T','E','S','V')`, and that actually excludes temporary tables whose value in `TABLETYPE` column is 'X'. As a result, if we call show_create_table on a global temporary table, it runs through but return no SQL create statement.

The new code would get the `TableDescriptor` of a global temporary table just as a base table and eventually reaches line 2225 and below. But `CREATE (LOCAL|GLOBAL) TEMPORARY TABLE` statement has two special terms `[NOLOGGING | ON COMMIT PRESERVE ROWS]`. So maybe we should have an extra branch for temporary tables and check these two properties. Or we can keep the old behavior and silently skip for temporary tables.

Reply:
Thanks @ascend1 ! I think it makes sense to support show create table for temp table also. I'll add the code to handle the options.

Reply:
I've added the code to support temporary table.